### PR TITLE
The awesome fancy box update which looks super epic, immersive and cool

### PR DESCRIPTION
**Pull request by alsoandanswer**
- rscadd: "The cookie snack now contains cookies, like the mint and gum packs. They've also been added to the rations box selection."
  - rscadd: "Gum packs, mint packs and the cookie snack from the vending machines visibly open when clicked on."
  - rscadd: "Gum packs, mint packs and the cookie snack can be crumpled up when empty. If you really wanted to, you still can crumple it with things inside, so long as you're in harm intent."
  - rscadd: "Odd boxes, such as candle boxes, donut boxes, crayon boxes and egg boxes can be folded and unfolded from cardboard sheets."
  - rscadd: "Donut boxes, egg cartons and cigarette packets visibly open when clicked on. Alt-click to close."
  - rscadd: "Mints actually have a proper taste description now."

anyway, changed the path for fancy boxes to actually be boxes. which explains the large amount of files change. also because it's more convenient and i can piggyback on the folding functionality which saves a lot of copypasta. which also means fancy boxes can be folded into sheets, conveniently.
I also tacked on a trash functionality on the sheet making one, which is pretty clean if i do say so myself.

Also I improved the examine text.

here's a gif to the new graphics stuff too.
qsPyDxEKlX

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 1]
```diff
@@ -522,6 +525,16 @@
 	W.update_icon()
 	return
 
+/obj/item/storage/box/fancy/matches/update_icon()
+	.=..()
+	if(opened)
+		if(contents.len <= 0)
```
Comment (mikomyazaki) [inline code/game/objects/items/weapons/storage/boxes.dm:531]: list.len can never be less than 0.

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 2]
```diff
@@ -50,12 +50,11 @@
...
 /obj/item/storage/box/fancy/examine(mob/user)
-	if(!..(user, 1))
-		return
+	..()
 	if(contents.len <= 0)
-		to_chat(user, SPAN_NOTICE("There are no [src.icon_type]s left in the [src.storage_type]."))
+		to_chat(user, "There are no [src.icon_type]s left in the [src.storage_type].")
 	else
-		to_chat(user, SPAN_NOTICE("There [src.contents.len == 1 ? "is" : "are"] <b>[src.contents.len]</b> [src.icon_type]\s left in the [src.storage_type]."))
+		to_chat(user, "There [src.contents.len == 1 ? "is" : "are"] <b>[src.contents.len]</b> [src.icon_type]\s left in the [src.storage_type].")
```
Comment (mikomyazaki) [inline code/game/objects/items/weapons/storage/fancy.dm:57]:
```suggestion
		to_chat(user, "There [src.contents.len == 1 ? "is" : "are"] <b>[src.contents.len]</b> [src.icon_type]\s left in \the [src.storage_type].")
```

File: code/modules/mob/living/silicon/robot/robot_helpers.dm [hunk 3]
```diff
@@ -47,4 +47,4 @@
 	stored_module.loc = module
 	if(client)
 		client.screen -= stored_module
-	contents -= stored_module
\ No newline at end of file
+	contents -= stored_module
```
Comment (mikomyazaki) [inline code/modules/mob/living/silicon/robot/robot_helpers.dm:50]: You could just not commit this file if there's no changes. Doesn't matter I guess.

File: code/modules/clothing/suits/jobs.dm [hunk 4]
```diff
@@ -128,7 +128,7 @@ obj/item/clothing/suit/apron/overalls/blue
 	blood_overlay_type = "coat"
 	contained_sprite = TRUE
 	body_parts_covered = UPPER_TORSO|ARMS
-	allowed = list(/obj/item/tank/emergency_oxygen, /obj/item/device/flashlight,/obj/item/gun/energy,/obj/item/gun/projectile,/obj/item/ammo_magazine,/obj/item/ammo_casing,/obj/item/melee/baton,/obj/item/handcuffs,/obj/item/storage/fancy/cigarettes,/obj/item/flame/lighter,/obj/item/device/taperecorder)
+	allowed = list(/obj/item/tank/emergency_oxygen, /obj/item/device/flashlight,/obj/item/gun/energy,/obj/item/gun/projectile,/obj/item/ammo_magazine,/obj/item/ammo_casing,/obj/item/melee/baton,/obj/item/handcuffs,/obj/item/storage/box/fancy/cigarettes,/obj/item/flame/lighter,/obj/item/device/taperecorder)
```
Comment (mikomyazaki) [inline code/modules/clothing/suits/jobs.dm:131]: Could put each of these on their own line for readability. Here and elsewhere.

File: code/modules/mob/living/silicon/robot/robot_helpers.dm (hunk 3 above)
Comment (alsoandanswer) [reply code/modules/mob/living/silicon/robot/robot_helpers.dm:50]: Motherfu-

**Pull request merged**